### PR TITLE
Revise “Configure a Pod to Use a ConfigMap”

### PR DESCRIPTION
**Pull request by sftim**
- style tidying
- clean up references to old Kubernetes versions
- wording tweaks

/kind cleanup
/language en

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 1]
```diff
@@ -623,7 +626,7 @@ basis. The [Secrets](/docs/concepts/configuration/secret/#using-secrets-as-files
...
+When a ConfigMap already being consumed in a volume is updated, projected keys are eventually updated as well. The lubelet is checking whether the mounted ConfigMap is fresh on every periodic sync. However, it is using its local TTL-based cache for getting the current value of the ConfigMap. As a result, the total delay from the moment when the ConfigMap is updated to the moment when new keys are projected to the pod can be as long as kubelet sync period (1 minute by default) + TTL of ConfigMaps cache (1 minute by default) in kubelet. You can trigger an immediate refresh by updating one of the Pod's annotations.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:629]: Fix me: "The lubelet"

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 2]
```diff
@@ -589,11 +594,9 @@ SPECIAL_TYPE
 If there are some files in the `/etc/config/` directory, they will be deleted.
 {{< /caution >}}
 
-{{< note >}}
-Text data is exposed as files using the UTF-8 character encoding. To use some other character encoding, use binaryData.
-{{< /note >}}
+Text data is exposed as files using the UTF-8 character encoding. To use some other character encoding, use `binaryData`.
 
-### Add ConfigMap data to a specific path in the Volume
+### Add ConfigMap data to a specific path in the volume
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:599]: This is referring to the `path` field in the Volume configuration?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 3]
```diff
@@ -377,21 +375,27 @@ configMapGenerator:
 EOF
 ```
 
-Apply the kustomization directory to create the ConfigMap object.
+Apply the kustomization directory to create the ConfigMap:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:378]: Should ConfigMap use code formatting?

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-24420--kubernetes-io-master-staging.netlify.app/docs/tasks/configure-pod-container/configure-pod-configmap/

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:629]: Ah yes that sounds like something else entirely.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:599]: The way I understand it, it's about telling the kubelet to write data into a mounted filesystem based on what's in a given ConfigMap.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:378]: Probably - would it be OK to save that update for a follow-up PR?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:599]: OK

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 3 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:378]: 👍

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 4]
```diff
@@ -304,12 +301,12 @@ For example:
 kubectl create configmap game-config-3 --from-file=game-special-key=configure-pod-container/configmap/game.properties
 ```
 
-would produce the following ConfigMap:
+would produce a ConfigMap than you can then fetch (to check):
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:304]: ? nit: s/than/that/

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 5]
```diff
@@ -449,86 +455,85 @@ configMapGenerator:
...
- ```
+Create the ConfigMap:
 
-* Use `envFrom` to define all of the ConfigMap's data as container environment variables. The key from the ConfigMap becomes the environment variable name in the Pod.
+```shell
+kubectl create -f https://kubernetes.io/examples/configmap/configmap-multikeys.yaml
+```
 
- {{< codenew file="pods/pod-configmap-envFrom.yaml" >}}
+Use `envFrom` to define all of the ConfigMap's data as container environment variables. The key from the ConfigMap becomes the environment variable name in the Pod.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:526]: nit: This line could be wrapped.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 6]
```diff
@@ -550,11 +555,11 @@ produces the following output in the `test-container` container:
 very charm
 ```
 
-## Add ConfigMap data to a Volume
+## Add ConfigMap data to a volume
 
-As explained in [Create ConfigMaps from files](#create-configmaps-from-files), when you create a ConfigMap using ``--from-file``, the filename becomes a key stored in the `data` section of the ConfigMap. The file contents become the key's value.
+As explained in [Create ConfigMaps from files](#create-configmaps-from-files), when you create a ConfigMap using `--from-file`, the filename becomes a key stored in the `data` section of the ConfigMap. The file contents become the key's value.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:560]: Could wrap this line

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 7]
```diff
@@ -623,7 +626,7 @@ basis. The [Secrets](/docs/concepts/configuration/secret/#using-secrets-as-files
...
+When a ConfigMap already being consumed in a volume is updated, projected keys are eventually updated as well. The kubelet is checking whether the mounted ConfigMap is fresh on every periodic sync. However, it is using its local TTL-based cache for getting the current value of the ConfigMap. As a result, the total delay from the moment when the ConfigMap is updated to the moment when new keys are projected to the pod can be as long as kubelet sync period (1 minute by default) + TTL of ConfigMaps cache (1 minute by default) in kubelet. You can trigger an immediate refresh by updating one of the Pod's annotations.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:629]: nit: Would be good to reword this paragraph. OK to leave to another PR.
The kubelet checks whether the mounted ...

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 8]
```diff
@@ -638,10 +641,10 @@ A container using a ConfigMap as a [subPath](/docs/concepts/storage/volumes/#usi
...
+ConfigMaps should reference properties files, not replace them. Think of the ConfigMap as representing something similar to the Linux `/etc` directory and its contents. For example, if you create a Kubernetes [volume](/docs/concepts/storage/volumes/) from a ConfigMap, each data item in the ConfigMap is represented by an individual file in the volume.
 {{< /note >}}
 
-The ConfigMap's `data` field contains the configuration data. As shown in the example below, this can be simple -- like individual properties defined using `--from-literal` -- or complex -- like configuration files or JSON blobs defined using `--from-file`.
+The ConfigMap's `data` field contains the configuration data. As shown in the example below, this can be simple — like individual properties defined using `--from-literal`, or complex — like configuration files or JSON blobs defined using `--from-file`.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:647]: nit: I'd like the lines wrapped.

**Comment by sftim** [comment]
I won't pick this up 'til I'm back from vacation, so marking as WIP again.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 9]
```diff
@@ -8,44 +8,53 @@ card:
...
 
 
 <!-- steps -->
 
 
 ## Create a ConfigMap
-You can use either `kubectl create configmap` or a ConfigMap generator in `kustomization.yaml` to create a ConfigMap. Note that `kubectl` starts to support `kustomization.yaml` since 1.14.
+You can use either `kubectl create configmap` or a ConfigMap generator in `kustomization.yaml` to create a ConfigMap.
 
 ### Create a ConfigMap Using kubectl create configmap
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:30]: nit: Create a ConfigMap using kubectl

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 10]
```diff
@@ -8,44 +8,53 @@ card:
...
 
 You can use [`kubectl describe`](/docs/reference/generated/kubectl/kubectl-commands/#describe) or
 [`kubectl get`](/docs/reference/generated/kubectl/kubectl-commands/#get) to retrieve information
 about a ConfigMap.
 
 #### Create ConfigMaps from directories
 
-You can use `kubectl create configmap` to create a ConfigMap from multiple files in the same directory. When you are creating a ConfigMap based on a directory, kubectl identifies files whose basename is a valid key in the directory and packages each of those files into the new ConfigMap. Any directory entries except regular files are ignored (e.g. subdirectories, symlinks, devices, pipes, etc).
+You can use `kubectl create configmap` to create a ConfigMap from multiple files in the same directory. When you are
+creating a ConfigMap based on a directory, kubectl identifies files whose basename is a valid key in the directory
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:55]: nit: kubectl should be formatted as code (line 55), `kubectl`?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 11]
```diff
@@ -61,7 +70,9 @@ wget https://kubernetes.io/examples/configmap/ui.properties -O configure-pod-con
 kubectl create configmap game-config --from-file=configure-pod-container/configmap/
 ```
 
-The above command packages each file, in this case, `game.properties` and `ui.properties` in the `configure-pod-container/configmap/` directory into the game-config ConfigMap. You can display details of the ConfigMap using the following command:
+The above command packages each file, in this case, `game.properties` and `ui.properties` in the
+`configure-pod-container/configmap/` directory into the game-config ConfigMap. You can display details of the
+ConfigMap using the following command:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:75]: nit: Should `game-config` use code formatting (name of ConfigMap)?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 4 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:318]: ?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 12]
```diff
@@ -8,44 +8,53 @@ card:
 ---
 
 <!-- overview -->
-ConfigMaps allow you to decouple configuration artifacts from image content to keep containerized applications portable. This page provides a series of usage examples demonstrating how to create ConfigMaps and configure Pods using data stored in ConfigMaps.
-
+{{< glossary_tooltip text="ConfigMaps" term_id="configmap" >}} allow you to decouple
+configuration data from container image artifacts to keep containerized applications
+portable. This page provides a series of usage examples demonstrating how to create
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:13]: There are several changes in this page that wrap the text. Is it preferred to wrap the text (say 80 chars) or
leave the longer lines as is?

Comment (tengqm) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:13]: We should encourage wrapping long lines because git works by comparing text line by line, so does github.com.
Wrapping long lines help identify changes, provide review suggestions, track upstream (en) changes when localizing a page.
Wrapping long lines at 80 characters boundary or 120 characters boundary? That doesn't matter too much. Maybe 80 characters is too restrictive or old fashioned?
When I'm editing a paragraph using vim, I always end by selecting at least one character in a paragraph and then press Q. The whole paragraph is well wrapped.

Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:13]: Ok, so what I could do is add a first commit that wraps long lines.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 4 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:318]: @sftim, The changes seem fine. Do you have time to fix line 318?
/lgtm

**Comment by sftim** [comment]
Revised!

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 11 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md]: I think it's clearer to use italics / emphasis. Not sure what the style guide thinks but _I_ want a way for readers to tell `name`s from `kind`s.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 12 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:13]: Done!

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 13]
```diff
@@ -8,66 +8,80 @@ card:
...
+`ConfigMap`s and configure `Pod`s using data stored in `ConfigMap`s.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:14]: @sftim, In reading the page again, the mix of fonts does not look quite right:
`ConfigMap`s and `Pod`s.
What do you think?

Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:14]: I'm minded to go with this for now, but use the poor rendering that comes from following the current style guide to support a case for reverting the style guide so that we _don't_ write API object kinds in code style.

The thing you configure to use a ConfigMap _is_ a Pod, the API object. That's true no matter what font you write it in!

Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:14]: OK, though this could introduce more updates.
You could leave the existing text (ConfigMaps and configure Pods) and instead create an issue to modify the style guide explaining the problems.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 14]
```diff
@@ -8,66 +8,80 @@ card:
...
+## Creating ConfigMaps {#create-a-configmap}
+To create a `ConfigMap`, you can use `kubectl create configmap`, or a `ConfigMap`
+generator in `kustomization.yaml`.
 
-### Create a ConfigMap Using kubectl create configmap
-
-Use the `kubectl create configmap` command to create ConfigMaps from [directories](#create-configmaps-from-directories), [files](#create-configmaps-from-files), or [literal values](#create-configmaps-from-literal-values):
+### Create a ConfigMap using `kubectl create` {#create-a-configmap-using-kubectl-create-configmap} 
+Use the `kubectl create configmap` command to create `ConfigMap`s
+from [directories](#create-configmaps-from-directories),
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:32]: nit: here you could use the singular, " ... to create a `ConfigMap` from  ...

Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:32]: Check the anchor, `#create-configmaps-from-directories),`

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 15]
```diff
@@ -8,66 +8,80 @@ card:
...
 wget https://kubernetes.io/examples/configmap/ui.properties -O configure-pod-container/configmap/ui.properties
 
-# Create the configmap
+# Create the ConfigMap
 kubectl create configmap game-config --from-file=configure-pod-container/configmap/
 ```
 
-The above command packages each file, in this case, `game.properties` and `ui.properties` in the `configure-pod-container/configmap/` directory into the game-config ConfigMap. You can display details of the ConfigMap using the following command:
+The above command packages each file, in this case, `game.properties` and `ui.properties`
+in the `configure-pod-container/configmap/` directory into the _game-config_ `ConfigMap`.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:77]: nit: Lines 76-77 could move to line 61 or just before the command instructions? This avoids telling the reader
to refer to "the above command"

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 16]
```diff
@@ -201,34 +221,33 @@ allow.textmode=true
 how.nice.to.look=fairlyNice
 ```
 
-Use the option `--from-env-file` to create a ConfigMap from an env-file, for example:
+Use the option `--from-env-file` to create a `ConfigMap` from an env-file.
+env-files contain a list of environment variables.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:225]: Could replace, "env-files" with something like:
"An environment variables file or _env-file_ contains a list of environment variables."

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 17]
```diff
@@ -201,34 +221,33 @@ allow.textmode=true
...
-# Download the sample files into `configure-pod-container/configmap/` directory
-wget https://kubernetes.io/examples/configmap/game-env-file.properties -O configure-pod-container/configmap/game-env-file.properties
+If the value includes quotation marks, those are copied into the ConfigMap verbatim.
 
-# The env-file `game-env-file.properties` looks like below
-cat configure-pod-container/configmap/game-env-file.properties
-enemies=aliens
-lives=3
-allowed="true"
+This example creates a `ConfigMap` from an env-file:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:235]: nit: two properties files?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 18]
```diff
@@ -288,28 +309,30 @@ data:
   textmode: "true"
 ```
 
-#### Define the key to use when creating a ConfigMap from a file
+#### Define the key to use when creating a `ConfigMap` from a file
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:312]: I would avoid using code style in a heading.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 19]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
-#### Generate ConfigMaps from Literals
-To generate a ConfigMap from literals `special.type=charm` and `special.how=very`,
-you can specify the ConfigMap generator in `kustomization.yaml` as
-```shell
-# Create a kustomization.yaml file with ConfigMapGenerator
-cat <<EOF >./kustomization.yaml
+### Generate ConfigMaps using literal data {#generate-configmaps-from-literals}
+
+This example shows you how to creat a `ConfigMap` from two literal key/value pairs:
+`special.type=charm` and `special.how=very`, using Kustomize and kubectl. To achieve
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:476]: nit: `kustomize` and `kubectl`?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 20]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
+this, you can specify the `ConfigMap` generator. Create (or replace)xi
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:477]: Check "Create (or replace)xi"?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 21]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
 ```shell
 kubectl apply -k .
+```
+```
 configmap/special-config-2-c92b5mmcf2 created
 ```
 
+---
+
+Now that you have learned to define `ConfigMap`s, you can move on to the next
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:499]: Just noting that you could format as, ConfigMaps ...

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 22]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
-    ```shell
-    kubectl create configmap special-config --from-literal=special.how=very
-    ```
+   ```shell
+   kubectl create configmap special-config --from-literal=special.how=very
+   ```
 
-2.  Assign the `special.how` value defined in the ConfigMap to the `SPECIAL_LEVEL_KEY` environment variable in the Pod specification.
+2. Assign the `special.how` value defined in the ConfigMap to the `SPECIAL_LEVEL_KEY`
+   environment variable in the `Pod` specification. Here's the manifest you'll use.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:516]: nit: Here's the manifest you'll use:

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 23]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
 
- ```shell
- kubectl create -f https://kubernetes.io/examples/pods/pod-single-configmap-env-variable.yaml
- ```
+   ```shell
+   kubectl create -f https://kubernetes.io/examples/pods/pod-single-configmap-env-variable.yaml
+   ```
 
-   Now, the Pod's output includes environment variable `SPECIAL_LEVEL_KEY=very`.
+   The `Pod`'s output includes environment variable `SPECIAL_LEVEL_KEY=very`.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:526]: The Pod ...

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 24]
```diff
@@ -430,111 +464,127 @@ configMapGenerator:
...
 
 ### Define container environment variables with data from multiple ConfigMaps
 
- * As with the previous example, create the ConfigMaps first.
-
-   {{< codenew file="configmap/configmaps.yaml" >}}
-
-   Create the ConfigMap:
+As with the previous example, create the `ConfigMap`s first. Here is a manifest for
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:530]: Edit: ... create the ConfigMaps first.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 25]
```diff
@@ -8,66 +8,80 @@ card:
...
+[files](#create-configmaps-from-files), or
+[literal values](#create-configmaps-from-literal-values):
 
 ```shell
 kubectl create configmap <map-name> <data-source>
 ```
 
-where \<map-name> is the name you want to assign to the ConfigMap and \<data-source> is the directory, file, or literal value to draw the data from.
-The name of a ConfigMap object must be a valid
+where \<map-name> is the name you want to assign to the `ConfigMap`, and
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:40]: nit: format as `<map-name>` and `<data-source>`?

**Comment by sftim** [comment]
Even though this has a few nits, it might be OK to merge as-is and leave the defects to become good-first-issues.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 13 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:14]: ?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 26]
```diff
@@ -8,66 +8,80 @@ card:
...
+{{< include "task-tutorial-prereqs.md" >}}
 
 
 <!-- steps -->
 
 
-## Create a ConfigMap
-You can use either `kubectl create configmap` or a ConfigMap generator in `kustomization.yaml` to create a ConfigMap. Note that `kubectl` starts to support `kustomization.yaml` since 1.14.
+## Creating ConfigMaps {#create-a-configmap}
+To create a `ConfigMap`, you can use `kubectl create configmap`, or a `ConfigMap`
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:27]:
```suggestion
To create a `ConfigMap`, you can use `kubectl create configmap`, or a ConfigMap resource generator
```

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 18 above)
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:312]:
```suggestion
#### Define the key to use when creating a ConfigMap from a file
```

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 27]
```diff
@@ -330,21 +353,23 @@ data:
     secret.code.lives=30
 ```
 
-#### Create ConfigMaps from literal values
+### Create ConfigMaps from literal values
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:356]:
```suggestion
### Create ConfigMap objects from literal values
```

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 19 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:476]: code format `kustomize` and `kubectl`?

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 20 above)
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:477]:
```suggestion
this, you can specify the `ConfigMap` generator. Create (or replace)
```

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 14 above)
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:32]:
```suggestion
from [directories](#create-a-configMap-from-a-director),
```

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 28]
```diff
@@ -8,66 +8,80 @@ card:
...
+[literal values](#create-configmaps-from-literal-values):
 
 ```shell
 kubectl create configmap <map-name> <data-source>
 ```
 
-where \<map-name> is the name you want to assign to the ConfigMap and \<data-source> is the directory, file, or literal value to draw the data from.
-The name of a ConfigMap object must be a valid
+where \<map-name> is the name you want to assign to the `ConfigMap`, and
+\<data-source> is the directory, file, or literal value to draw the data from.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:41]:
```suggestion
`<data-source>` is the directory, file, or literal value to draw the data from.
```

Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:41]: ?

**Comment by kbhawkey** [comment]
Hi @sftim .
I read through the changes again. Most of the changes are fine and improve the content.
There are remaining questions about the code formatting/usage of the common terms/API resource type names.
Patterns I see in the page:
Pod, Pods, ConfigMap, ConfigMap objects, ConfigMaps, `Pod`, `Pod`s, `ConfigMap`s, `ConfigMap`. 
Before this can merge, I suggest you fix any links that do not resolve and clean up any typos.
I don't think I would update an existing type name with code formatting if the name is plural (`ConfigMap`s).
You can generally refer to the Pod type as pods. The ConfigMap object presents a different problem.
Hope this helps move this PR forward.

**Comment by sftim** [comment]
(needs some work)

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 26 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:27]: Addressed.

File: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md (hunk 27 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md:370]: We should take care to preserve existing hyperlinks. Also I think I prefer the original heading anyway here.

**Comment by shannonxtreme** [comment]
@sftim whenever you do a rebase on this, please open it up for review. I'm happy to scan it and see if any major changes are needed before a merge. 

/label refactor
/remove-lifecycle rotten

**Comment by jimangel** [comment]
@sftim this is k/website's second oldest open PR. Shall we close it (I would like to see the content merge) or do you need help getting it out of Draft?

**Comment by sftim** [comment]
This one I'll close and aim to revisit.
/close
